src/types.jl

```diff
@@ -149,12 +507,552 @@ for Model in [:NeuralNetworkRegressor, :MultitargetNeuralNetworkRegressor]
+
+- `chain`: The trained "chain", or series of layers, functions, and activations which
+  make up the neural network.
+
+
+# Report
+
+The fields of `report(mach)` are:
+
+- `training_losses`: The history of training losses, a vector containing the history of
+  all the losses during training. The first element of the vector is the initial
+  penalized loss. After the first element, the nth element corresponds to the loss of
+  epoch n-1.
+
+# Examples
+
+In this example we build a regression model using a toy dataset.
+```julia
+using MLJ
+using MLJFlux
+using Flux
+using Plots
+using MLJBase: augment_X
+```
+First, we generate some data:
+```julia
+X = augment_X(randn(10000, 8), true);
+θ = randn((9,2));
+y = X * θ;
+X = MLJ.table(X)
+y = MLJ.table(y)
+
+schema(y)
+schema(X)
+```
+Lets also make a test set:
+```julia
+(X, Xtest), (y, ytest) = partition((X, y), 0.7, multi=true);
+```
+Next, we can define a `builder`. In the following macro call, `n_in` is the number of expected input features, and rng is a RNG. `init` is the function used to generate the random initial weights of the network.
+```julia
+builder = MLJFlux.@builder begin
+  init=Flux.glorot_uniform(rng)
+  Chain(Dense(n_in, 64, relu, init=init),
+        Dense(64, 32, relu, init=init),
+        Dense(32, 1, init=init))
+end
+```
+Finally, we can define the model!
+```julia
+MultitargetNeuralNetworkRegressor = @load MultitargetNeuralNetworkRegressor
+  model = MultitargetNeuralNetworkRegressor(builder=builder,
+                                 rng=123,
+                                 epochs=20)
+```
+For our neural network, since different features likely have different scales, if we do not standardize the network may be implicitly biased towards features with higher magnitudes, or may have [saturated neurons](https://www.informit.com/articles/article.aspx%3fp=3131594&seqNum=2)  and not train well. Therefore, standardization is key!
+```julia
+pipe = Standardizer |> TransformedTargetModel(model, target=Standardizer)
+```
+If we fit with a high verbosity (>1), we will see the losses during training. We can also see the losses in the output of `report(mach)`
+
+```julia
+mach = machine(pipe, X, y)
+fit!(mach, verbosity=2)
+
+# first element initial loss, 2:end per epoch training losses
+report(mach).transformed_target_model_deterministic.training_losses
+
+```
+
+## Experimenting with learning rate
+
+We can visually compare how the learning rate affects the predictions:
+```julia
+plt = plot()
+
+rates = 10. .^ (-5:0)
+
+foreach(rates) do η
+  pipe.transformed_target_model_deterministic.model.optimiser.eta = η
+  fit!(mach, force=true, verbosity=0)
+  losses =
+      report(mach).transformed_target_model_deterministic.model.training_losses[3:end]
+  plot!(1:length(losses), losses, label=η)
+end
+plt #!md
+
+savefig(joinpath("assets", "learning_rate.png"))
+
+
+pipe.transformed_target_model_deterministic.model.optimiser.eta = 0.0001
+
+```
+
```

Comment:
As for vanilla regressor, let's wrap up here. 

One extra issue is that MLJ has no built-in multi-target measures. But we can define a custom one in one line. So here's a suggestion to wrap up:

---

With the learning rate fixed we now compute a CV estimate of the performance (using all data bound to `mach`) and compare this with performance on the test set:

```julia
# custom MLJ loss:
multi_loss(yhat, y) = l2(MLJ.matrix(yhat), MLJ.matrix(y)) |> mean

# CV estimate, based on `(X, y)`:
evaluate!(mach, resampling=CV(nfolds=5), measure=multi_loss)

# loss for `(Xtest, test)`:
fit!(mach)
yhat = predict(mach, Xtest)
multi_loss(yhat, y)
```